botbot_plugins/plugins/vote.py

```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8 -*-
+
+import json
+import re
+from functools import wraps
+from collections import namedtuple
+
+from .. import config
+from ..base import BasePlugin
+from ..decorators import (listens_to_all, listens_to_command,
+                          listens_to_regex_command)
+
+ERROR_MESSAGES = {
+    "no_vote_started": u"No vote has been started. Use the “startvote” command to do so.",
+    "voting_already_running": u"{author}: There’s already a vote going on. Use the “endvote” command to end it before starting a new one.",
+    "invalid_option": u"“{option}” is not a valid option."
+}
+INFO_MESSAGES = {
+    "voting_started": u"Voting has started.",
+    "voting_started_name": u"Voting has started for proposal “{name}”.",
+    "voting_ended": u"Voting has ended.",
+    "voting_ended_name": u"Voting has ended for proposal “{name}”."
+}
+
+
+def get_section_template(name, for_, against, username_sep):
+    template = (u"[{}(+{}, -{}): "
+                .format(name, len(for_), len(against)))
+
+    if for_:
+        template += username_sep.join(for_)
+    if for_ and against:
+        template += u"; "
+    if against:
+        template += username_sep.join(
+            [u"-" + username for username in against])
+    template += u"]"
+    return template
+
+
+COUNTVOTE_TEMPLATES = {
+    "default_for_template": u"[+{number_for}: {users}]",
+    "default_against_template": u"[-{number_against}: {users}]",
+    "abstain_template": u"[\\{number_abstain}: {users}]",
+    "section_template": get_section_template
+}
+
+FOR_NAME = "for"
+AGAINST_NAME = "against"
+ABSTAIN_NAME = "abstain"
+OPPOSING_VOTE = {
+    FOR_NAME: AGAINST_NAME,
+    AGAINST_NAME: FOR_NAME
+}
+VOTE_SHORTCUT_TO_NAME = {
+    "": FOR_NAME,
+    "+": FOR_NAME,
+    "-": AGAINST_NAME
+}
```

Comment:
Move all of this up above get_section_template(), keep the constants together,

Reply:
Can't do that since `COUNTVOTE_TEMPLATES` uses `get_section_template`.